src/prefect/events/related.py

```diff
@@ -46,13 +46,19 @@ def tags_as_related_resources(tags: Iterable[str]) -> List[RelatedResource]:
 def object_as_related_resource(kind: str, role: str, object: Any) -> RelatedResource:
     resource_id = f"prefect.{kind}.{object.id}"
 
-    return RelatedResource.model_validate(
-        {
-            "prefect.resource.id": resource_id,
-            "prefect.resource.role": role,
-            "prefect.resource.name": object.name,
-        }
-    )
+    labels = {
+        "prefect.resource.id": resource_id,
+        "prefect.resource.role": role,
+        "prefect.resource.name": object.name,
+    }
+
+    if role == "deployment":
+        from prefect.client.schemas.responses import DeploymentResponse
+
+        if isinstance(object, DeploymentResponse):
+            return object.as_related_resource("deployment")
```

Comment:
What do you think about going full duck-typing here?
```suggestion
    if obj_as_related_resource := getattr(object, "as_related_resource", None):
        return obj_as_related_resource()
```

Reply:
I think it's freaking FANTASTIC, will do